test/e2e/common/secrets.go

```diff
@@ -313,3 +285,88 @@ func doSecretE2EWithMapping(f *framework.Framework, mode *int32) {
 
 	f.TestContainerOutput("consume secrets", pod, 0, expectedOutput)
 }
+
+//Existence of a second secret with same name and different namespace shouldn't affect the mount
+func doSecretE2EForMountNameSpaceIssue(f *framework.Framework, defaultMode *int32) {
+	var (
+		name            = "secret-test-" + string(uuid.NewUUID())
+		volumeName      = "secret-volume"
+		volumeMountPath = "/etc/secret-volume"
+		namespace1      = f.Namespace
+		namespace2      *api.Namespace
+		err             error
+	)
+
+	if namespace2, err = f.CreateNamespace("secret-namespace", nil); err != nil {
+		framework.Failf("unable to create new namespace %s: %v", namespace2.Name, err)
+	}
+
+	secret1, secret2 := secretForTest(namespace1.Name, name), secretForTest(namespace2.Name, name)
+
+	By(fmt.Sprintf("Creating secrets with name %s and namespaces %s %s", name, namespace1.Name, namespace2.Name))
+
+	if secret1, err = f.ClientSet.Core().Secrets(namespace1.Name).Create(secret1); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret1.Name, err)
+	}
+
+	if secret2, err = f.ClientSet.Core().Secrets(namespace2.Name).Create(secret2); err != nil {
+		framework.Failf("unable to create test secret %s: %v", secret2.Name, err)
+	}
+
+	secret2.Data = map[string][]byte{
+		"this_should_not_match_previous_info": []byte("similarly_this_should_not_match_previous_info\n"),
+	}
+
```

Comment:
Avoiding code duplication is generally a good idea for maintainability (if there is a bug, you only have to fix it in one place). That said, sometimes avoiding code duplication makes the single block of code much more difficult to understand. So it's a trade-off between maintainability and readability.

In this case, I agree with @rata, deduplicating the code here would be beneficial. It would make this code easier to maintain without impacting readability. The new test could create a 2nd namespace and secret before calling the common method `doSecretE2EWithMapping(...)`. And that method can be modified to take namespace and secret name as parameters instead of hardcoding the same namespace and secret name for all callers.